packages/admin/cms-admin/src/common/extractImportRichtextStyles.ts

```diff
@@ -0,0 +1,79 @@
+import { RawDraftContentBlock, RawDraftInlineStyleRange } from "draft-js";
+
+export const extractRichtextStyles = (block: RawDraftContentBlock): string => {
+    let text = "";
+
+    // As multiple inline styles with same offset and length are possible, items with same offset and length are removed
+    const filteredInlineStyles = block.inlineStyleRanges
+        .map((inlineStyle) => ({ start: inlineStyle.offset, end: inlineStyle.offset + inlineStyle.length, type: "inlineStyle" }))
+        .reduce((previous: { start: number; end: number; type: string }[], current) => {
+            if (!previous.some((style: { start: number; end: number }) => style.start === current.start && style.end === current.end)) {
+                previous.push(current);
+            }
+            return previous;
+        }, []);
+
+    const combinedSettings = [
+        ...filteredInlineStyles,
+        ...block.entityRanges.map((entityRange) => ({
+            start: entityRange.offset,
+            end: entityRange.offset + entityRange.length,
+            type: "entityRange",
+        })),
+    ];
+
+    for (let i = 0; i < block.text.length + 1; i++) {
+        const startTags = combinedSettings.filter((setting) => setting.start === i);
+        const endTags = combinedSettings.filter((setting) => setting.end === i);
+
+        text += `${endTags.map((tag) => (tag.type === "inlineStyle" ? "</i>" : "</e>"))}${startTags.map((tag) =>
```

Comment:
Here you're mapping an array into a string without a join -> does that mean there's always only one element?